### PR TITLE
Fix build on Centos 7

### LINKED ISSUE #19: failed to build module on centos:7
```
[ 63%] Building C object src/CMakeFiles/merger.dir/compat/utils.c.o
[ 72%] Building C object src/CMakeFiles/merger.dir/merger/merger.c.o
/tmp/luarocks_tuple-merger-0.0.1-1-Cwe2jE/tuple-merger/src/merger/merger.c: In function 'luaT_merger_new_parse_sources':
/tmp/luarocks_tuple-merger-0.0.1-1-Cwe2jE/tuple-merger/src/merger/merger.c:306:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for (uint32_t i = 0; i < source_count; ++i) {
  ^
/tmp/luarocks_tuple-merger-0.0.1-1-Cwe2jE/tuple-merger/src/merger/merger.c:306:2: note: use option -std=c99 or -std=gnu99 to compile your code
gmake[5]: *** [src/CMakeFiles/merger.dir/merger/merger.c.o] Error 1
gmake[4]: *** [src/CMakeFiles/merger.dir/all] Error 2
gmake[3]: *** [all] Error 2
```

### PR DESCRIPTION
**Pull request by olegrok**
Initially I planed only CentOS 7 build. However finally I faced a problem with github actions.

Please, read description in commit messages.

Closes #19

### CODE REVIEW

**Review by tsafin — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,10 @@ set(CMAKE_MODULE_PATH
 
 include(cmake/utils.cmake)
 
+# Set CFLAGS/CXXFLAGS
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -std=gnu11")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Werror -std=gnu++11")
```
Comment (Totktonada) [inline CMakeLists.txt:16]: Ther is no C++ code here.

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,10 @@ set(CMAKE_MODULE_PATH
 
 include(cmake/utils.cmake)
 
+# Set CFLAGS/CXXFLAGS
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Werror -std=gnu11")
```
Comment (Totktonada) [inline CMakeLists.txt:15]: Is not gnu99 enough?

Comment (Totktonada) [inline CMakeLists.txt:15]: Maybe enable -Werror only on debug build, just like in tarantool?

File: .github/workflows/luarocks-build.yml [hunk 3]
```diff
@@ -65,7 +65,7 @@ jobs:
         run: |
           cd $GITHUB_WORKSPACE/tarantool/build
           cmake --build . --config $BUILD_TYPE --target install -- -j
-          echo "::set-env name=PATH::$GITHUB_WORKSPACE/share/bin:$PATH"
+          echo "PATH=$GITHUB_WORKSPACE/share/bin:$PATH" >> $GITHUB_ENV
```
Comment (Totktonada) [inline .github/workflows/luarocks-build.yml:68]: Don't know why extra path is needed, but I don't mind the change.

**Review by Totktonada — APPROVED** [review]
LGTM in general.

File: CMakeLists.txt (hunk 1 above)
Comment (olegrok) [reply CMakeLists.txt]: Removed

File: CMakeLists.txt (hunk 2 above)
Comment (olegrok) [reply CMakeLists.txt]: Yes, you are right. I tried to use "c99" and it failed. gnu99 works fine. Fixed. Thanks!

Comment (olegrok) [reply CMakeLists.txt]: Not sure it's a good idea. Tarantool has separate target for Debug build in CI but we don't have that for modules.

I don't see nothing bad to show and prevent as much warnings as possible.

File: .github/workflows/luarocks-build.yml (hunk 3 above)
Comment (olegrok) [reply .github/workflows/luarocks-build.yml:68]: Yes, I suggest fix CI in this way. Not sure we should rebuild Tarantool every time. One day we can change it to [setup-tarantool](https://github.com/tarantool/setup-tarantool)

File: CMakeLists.txt (hunk 1 above)
Comment (tsafin) [reply CMakeLists.txt]: Now, given @Totktonada has pointed that out, I start to worry about this option too. We do not have C++ code there (yet) and there is no point to enforce [awkward GNU extensions for C++](https://gcc.gnu.org/onlinedocs/gcc/C_002b_002b-Extensions.html) language at the moment when it's unnecessary.

Comment (tsafin) [reply CMakeLists.txt]: Sorry for the cofusion created - have not updated browser before sending comment. Now I see it's fixed

File: .github/workflows/luarocks-build.yml (hunk 3 above)
Comment (tsafin) [reply .github/workflows/luarocks-build.yml:68]: setup-tarantool is useful for using stable releases, due to caching involved. Here we need to compile against master, where caching would not be useful that much.

Comment (Totktonada) [reply .github/workflows/luarocks-build.yml:68]: However after https://github.com/tarantool/setup-tarantool/issues/9 it'll able to install latest master from a package. It should be faster than cloning and building tarantool.

File: CMakeLists.txt (hunk 2 above)
Comment (Totktonada) [reply CMakeLists.txt]: 1. You can run `cmake . -DCMAKE_BUILD_TYPE=Debug|Release|RelWithDebInfo && make VERBOSE=1` and see different compilation flags.
2. https://github.com/tarantool/tarantool/blob/efc30ccf8393a1aad19d4c17299842899d003caa/cmake/compiler.cmake#L372-L373 (it comes from this [commit](https://github.com/tarantool/tarantool/commit/7fa4be57aa540b995909150f4026d689c661b80d) and this [issue](https://bugs.launchpad.net/tarantool/+bug/805350)).

Comment (olegrok) [reply CMakeLists.txt]: Yes, we can, but I think it's more relevant for Tarantool with default Debug mode. But I'm not sure it's so relevant for modules.

E.g. in avro-schema:  https://github.com/tarantool/avro-schema/blob/3b2d264e1f7828dda034aa5cae30184f2b9adbfa/CMakeLists.txt#L21-L23

Comment (Totktonada) [reply CMakeLists.txt]: Proportionally to amount of code.

Default cmake flags for different build types are different.

**Pull request merged**